integration-test.Rmd

```diff
@@ -23,224 +13,16 @@ knitr::opts_chunk$set(
-
-Ensure the expected working directory.
-
-```{r}
-expect_equal(path_file(here()), "PACTA_analysis")
-```
-
-## `portfolio_name_ref_all <- "TestPortfolio_Input"`
-
-Ensure `portfolio_name_ref_all` takes the value "TestPortfolio_Input" in the files which name contains "web\_tool\_scripts".
-
-```{r}
-# What value is currently assigned to the variable `portfolio_name_ref_all`?
-show_pattern_in_file <- function(file, pattern) {
-  grep(pattern, readLines(file), value = TRUE)
-}
-
-pattern <- "set_portfolio-name-ref-all_working-location_and_web-parameters.R"
-(files <- dir_ls("deduplicate", regexp = pattern))
-
-this_pattern <- "portfolio_name_ref_all.*<-"
-matched <- map(files, show_pattern_in_file, pattern = this_pattern)
-walk(matched, writeLines)
-
-script_has_this_pattern <- grepl(this_pattern, matched)
-expect_true(any(script_has_this_pattern))
-```
-
-**NOTE: If the value of `portfolio_name_ref_all` comes from the user, we should provide an interface for the user to supply it -- so that the user needs not to change the source code.**
-
-## Configurations
-
-Ensure this configuration file exists:
-
-```{r}
-config_1 <- here(
-  "working_dir",
-  "10_Parameter_File",
-  "TestPortfolio_Input_PortfolioParameters.yml"
-)
-
-expect_true(file_exists(config_1))
-```
-
-```{r}
-look_into <- function(path, n = -1L) {
-  lines <- readLines(path, n, encoding = "UTF-8")
-  writeLines(lines)
-}
-
-look_into(config_1)
-```
-
-Ensure this other configuration file also exists:
-
-```{r}
-config_2 <- here("parameter_files", "WebParameters_2dii.yml")
-config_2_copy <- tempfile()
-fs::file_copy(config_2, config_2_copy)
-
-expect_true(file_exists(config_2))
-```
-
-Ensure the paths in the configuration file work both locally and on GitHub actions:
-
-```{r}
-make_config_portable <- function(config) {
-  lines <- readLines(config, encoding = "UTF-8")
-  lines <- make_paths_portable(lines)
-  writeLines(lines, config)
-
-  invisible(config)
-}
-
-make_paths_portable <- function(x) {
-  x %>%
-    root_field_path("project_location_ext", pattern = "PACTA_analysis") %>%
-    root_field_path("data_location_ext", pattern = "pacta-data") %>%
-    root_field_path("template_location", pattern = "create_interactive_report") %>%
-    root_field_path("stress_test_location", pattern = "StressTestingModelDev")
-}
-
-root_field_path <- function(x, field, pattern) {
-  parent <- path_dir(here())
-  value <- path(parent, extract_from(x, pattern))
-  sub(glue("({field}:[ ]?).*"), glue("\\1{value}/"), x)
-}
-
-extract_from <- function(x, pattern) {
-  line <- grep(pattern, x, value = TRUE)
-  sub(glue(".*({pattern}.*)"), "\\1", line)
-}
+## Introduction
 
-make_config_portable(config_2)
-```
+This document provides a reproducible example of how to run the so called "web tool". It is based on [these instructions](https://bit.ly/2RCRJn7). It is useful in three ways:
```

Comment:
I get "Access Denied" at this bit.ly URL

Reply:
@jacobvjk, can you grant access to @cjyetman to the google doc with the instructions? Maybe make it public?

Reply:
@cjyetman @maurolepore you should now both have access